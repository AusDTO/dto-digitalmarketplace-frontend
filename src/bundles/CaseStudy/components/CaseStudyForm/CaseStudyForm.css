p.referee-summary {
    margin: 1.2rem 0;
}