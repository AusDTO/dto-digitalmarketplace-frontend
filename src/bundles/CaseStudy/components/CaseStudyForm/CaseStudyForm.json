{
  "form": {
    "caseStudyForm": {
        "title": "Case Study Title",
        "opportunity": "The opportunity",
        "client": "The Client Name",
        "timeframe": "January 2016 â€” June 2016",
        "approach": "The approach",
        "outcome": [
            "Outcome 1",
            "Outcome 2"
        ],
        "links": [
            "http://gov.au/"
        ]
    }
  }
}
