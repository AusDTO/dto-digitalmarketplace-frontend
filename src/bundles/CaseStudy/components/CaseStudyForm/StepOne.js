import React from 'react';
import { connect } from 'react-redux';
import { Form } from 'react-redux-form';

import { required, validLinks } from '../../../../validators';

import Layout from '../../../../shared/Layout';

import BaseForm      from '../../../../shared/form/BaseForm';
import SubmitForm    from '../../../../shared/form/SubmitForm';
import ErrorBox      from '../../../../shared/form/ErrorBox';
import MultiInput    from '../../../../shared/form/MultiInput';
import Textarea      from '../../../../shared/form/Textarea';
import Textfield     from '../../../../shared/form/Textfield';

import { navigateStep } from '../../../../shared/reduxModules/form_options';

class StepOne extends BaseForm {

  componentDidMount() {
    this.props.dispatch(navigateStep(1));
    if (this.props.mounted) {
      this.refs['header'].focus();
    }
  }

  render() {
    const { 
      action,
      csrf_token,
      model,
      returnLink,
      mode,
      onClick,
      form
    } = this.props;

    return (
      <Layout>
        <header>
          <h1 tabIndex="-1" ref="header" aria-describedby="header-description">{mode === 'edit' ? 'Edit' : 'Add'} case study</h1>
          <p id="header-description">Show the range of skills and experience you can provide by completing the form below.</p>
        </header>
        <article role="main">
          <ErrorBox focusOnMount={true} model={model} />
          <Form model={model}
            action={action}
            method="post"
            id="casestudy__create"
            component={SubmitForm}
            valid={form.valid}
          >
            {csrf_token && (
              <input type="hidden" name="csrf_token" id="csrf_token" value={csrf_token} />
            )}
            <Textfield
              model={`${model}.title`}
              name="title"
              id="title"
              htmlFor="title"
              label="Give your case study a title"
              validators={{ required }}
              messages={{
                required: 'Title is required',
              }}
            />

            <Textfield
              model={`${model}.client`}
              name="client"
              id="client"
              htmlFor="client"
              label="Who was the client?"
              validators={{ required }}
              messages={{
                required: 'Client is required',
              }} />

            <Textfield
              model={`${model}.timeframe`}
              name="timeframe"
              id="timeframe"
              htmlFor="timeframe"
              label="What was the time frame?"
              description="For example,  January 2016 â€” June 2016"
              validators={{ required }}
              messages={{
                required: 'Timeframe is required',
              }}
            />

            <Textarea
              model={`${model}.opportunity`}
              name="opportunity"
              id="opportunity"
              controlProps={{ limit: 200 }}
              label="Outline the problem or opportunity"
              description="Describe the project goal and any relevant background information."
              messages={{
                required: 'You must outline the opportunity'
              }}
              validators={{ required }}
            />

            <Textarea
              model={`${model}.approach`}
              name="approach"
              id="approach"
              controlProps={{ limit: 200 }}
              label="Describe your approach"
              description="How did your capabilities and methods contribute to achieving the project goals?"
              messages={{
                required: 'You must outline your approach'
              }}
              validators={{ required }}
            />

            <MultiInput
              id="outcome"
              model={`${model}.outcome`}
              name="outcome"
              htmlFor="outcome"
              label="What was the outcome?"
              controlProps={{ defaultRows: 2 }}
              description="List the key achievements of this project."
              messages={{ required: 'You must provide at least one outcome.' }}
              validators={{ required }}
            />

            <MultiInput
              id="projectLinks"
              model={`${model}.projectLinks`}
              name="projectLinks"
              htmlFor="projectLinks"
              label="Project links"
              controlProps={{ defaultRows: 2 }}
              description="Link to any supporting material for your case study. This can include a case study on your  website, case study video or the live project."
              messages={{ validLinks: 'All links provided must begin with \'http\'' }}
              validators={{ validLinks }}
            />

            <input type="submit" value='Save &amp; Continue' role="button" onClick={onClick} />
          </Form>
          {returnLink && <a href={returnLink}>Return without saving</a>}
        </article>
      </Layout>
    )
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    ...ownProps
  }
}

export default connect(mapStateToProps)(StepOne);