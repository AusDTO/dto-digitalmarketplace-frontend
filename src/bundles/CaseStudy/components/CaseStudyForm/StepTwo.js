import React from 'react';
import { connect } from 'react-redux';
import { Form, Control } from 'react-redux-form';

import { required } from '../../../../validators';

import Layout from '../../../../shared/Layout';

import ErrorBox      from '../../../../shared/form/ErrorBox';
import StatefulError from '../../../../shared/form/StatefulError';
import Textfield     from '../../../../shared/form/Textfield';

import { navigateStep } from '../../../../shared/reduxModules/form_options';

import StepSidebar from './StepSidebar';

class StepTwo extends React.Component {

  componentDidMount() {
    this.props.dispatch(navigateStep(2));
  }

  render() {
    const { 
      action,
      csrf_token,
      model,
      returnLink,
      mode,
      caseStudyForm,
      onClick,
      attachNode,
      sidebarOptions
    } = this.props;

    // List form elements on step two, we dont want to send these fields twice.
    const exlcudedKeys = ['acknowledge', 'name', 'role', 'phone', 'email'];

    return (
      <Layout>
        <header>
          <h1>Reference for {caseStudyForm.title}</h1>
          <p>Show the range of skills and experience you can provide by completing the form below.</p>
        </header>
        <StepSidebar items={sidebarOptions} />
        <article role="main" className="content-main">
          <ErrorBox focusOnMount={true} model={model} />
          <Form model={model}
            action={action}
            method="post"
            id="casestudy__create"
            ref={attachNode}
          >
            {csrf_token && (
              <input type="hidden" name="csrf_token" id="csrf_token" value={csrf_token} />
            )}
            
            {/* 
              Print all previous fields as hidden fields for full form submit. Excluding fields on this page. 
              When/if this form moves away from full page submit and interacts with an API, this block can be removed.
            */}
            {Object.keys(caseStudyForm)
              .filter(key => exlcudedKeys.indexOf(key) === -1)
              .map((key, i) => {
                let field = caseStudyForm[key];
                if (Array.isArray(field)) {
                  return field.map((f, j) => (
                    <input key={`${i}${j}`} type="hidden" name={`${key}[]`} value={f} />
                  ))
                }
                return (
                  <input key={i} type="hidden" name={key} value={field} />
                )
              })
            }

            <Textfield
              model={`${model}.name`}
              name="name"
              id="name"
              htmlFor="name"
              label="Full name"
              description="Add the full name of the person who has agreed to be your reference."
              validators={{ required }}
              messages={{
                required: 'Full name is required',
              }}
            />

            <Textfield
              model={`${model}.role`}
              name="role"
              id="role"
              htmlFor="role"
              label="Role and place to work"
              description="For example,  Delivery Manager, Digital Transformation Agency."
              validators={{ required }}
              messages={{
                required: 'Role and place to work is required',
              }}
            />

            <Textfield
              model={`${model}.phone`}
              name="phone"
              id="phone"
              htmlFor="phone"
              label="Phone number"
              validators={{ required }}
              messages={{
                required: 'Phone number is required',
              }}
            />

            <Textfield
              model={`${model}.email`}
              name="email"
              id="email"
              htmlFor="email"
              label="Email"
              validators={{ required }}
              messages={{
                required: 'Email is required',
              }}
            />
            
            <div className="field">
              <Control.checkbox
                model={`${model}.acknowledge`}
                id="acknowledge"
                name="acknowledge"
                validators={{ required }}
              />
              <label htmlFor="acknowledge">I acknowledge this case study may be shared with registered buyers in the Digital Marketplace.</label>
              <StatefulError
                model={`${model}.acknowledge`}
                id="acknowledge"
                messages={{
                  required: 'You must acknowledge this case study will be shared.'
                }}
              />
            </div>

            <input type="submit" value={mode === 'add' ? 'Publish case study' : 'Update case study'} role="button" onClick={onClick} />
          </Form>
          {returnLink && <a href={returnLink}>Return without saving</a>}
        </article>
      </Layout>
    )
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    ...ownProps
  }
}

export default connect(mapStateToProps)(StepTwo);