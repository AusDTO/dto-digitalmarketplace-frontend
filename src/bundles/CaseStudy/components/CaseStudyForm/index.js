import CaseStudyForm from './CaseStudyForm'
export default CaseStudyForm
