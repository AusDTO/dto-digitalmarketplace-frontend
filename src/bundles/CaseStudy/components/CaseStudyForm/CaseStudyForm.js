import React from 'react';
import ReactDOM from 'react-dom';
import { connect } from 'react-redux';
import { actions } from 'react-redux-form';
import { Match, Miss } from 'react-router';

import StepOne from './StepOne';
import StepTwo from './StepTwo';

import { navigateStep } from '../../../../shared/reduxModules/form_options';


class CaseStudyForm extends React.Component {

  static propTypes = {
    action: React.PropTypes.string,
    csrf_token: React.PropTypes.string,
    form: React.PropTypes.object.isRequired,
    returnLink: React.PropTypes.string
  }

  static contextTypes = {
    router: React.PropTypes.shape({
      transitionTo: React.PropTypes.func.isRequired
    })
  }

  state = {
    localSubmitFailed: false
  }

  /**
   * We are calling this on `Will` instead of `Did` for server rendering purposes.
   * If there are formErrors available, set the appropriate errors and show them.
   * @return {void}
   */
  componentWillMount() {
    const { dispatch, formErrors, model, serverRender } = this.props;

    if (!formErrors) {
      return;
    }

    if (serverRender) {
      let errors = {};
      Object.keys(formErrors).forEach((key) => {
        errors[key] = {
          valid: false,
          errors: formErrors[key]
        }
      });
      dispatch(actions.setFieldsErrors(model, errors));
    }
  }

  componentDidMount() {
    const { dispatch, formErrors, model } = this.props;

    if (!formErrors) {
      return;
    }

    dispatch(actions.setSubmitFailed(model));
    dispatch(actions.setFieldsValidity(model, {}, { errors: true }));
  }

  attachNode(node) {
    this._form = ReactDOM.findDOMNode(node);
  }

  handleClick(e) {
    /**
     * FIXME
     * This is a workaround to complete a normal form submit
     */
    const { form, step, maxSteps, dispatch } = this.props;
    if (form.valid) {
      if (step < maxSteps) {
        e.preventDefault();

        const { router } = this.context;
        const nextStep = step + 1;

        dispatch(navigateStep(nextStep));
        router.transitionTo('/reference');

        this.setState({
          localSubmitFailed: false
        });
      } else {
        this._form.submit = this.refs.submittable.submit;
        this._form.submit();
      }
    } else {
      this.setState({
        localSubmitFailed: true
      });
    }
  }

  render() {
    const sidebarOptions = [
      { path: '/', label: 'Add case study' },
      { path: '/reference', label: 'Add reference' }
    ]
    const props = {
      ...this.props,
      sidebarOptions,
      attachNode: this.attachNode.bind(this),
      onClick: this.handleClick.bind(this)
    }

    return (
        <div>
          {/*FIXME: this form exists purely to steal its submit method.*/}
          <form ref="submittable" tabIndex="-1" style={{ display: "none" }} />
          <Match pattern="/" exactly render={(routerProps) => <StepOne {...routerProps} {...props} />} />
          <Miss render={(routerProps) => <StepOne {...routerProps} {...props} />} />
          <Match pattern="/reference" exactly render={(routerProps) => <StepTwo {...routerProps} {...props} />} /> 
        </div>
    )
  }
}

CaseStudyForm.defaultProps = {
  maxSteps: 1
}

const mapStateToProps = ({ forms, casestudy, form_options, caseStudyForm, options }) => {
  const form = forms.caseStudyForm.$form;
  return {
    form,
    caseStudyForm,
    maxSteps: 2,
    model: 'caseStudyForm',
    formErrors: form_options.errors,
    returnLink: casestudy.returnLink,
    ...form_options,
    ...options
  }
}

export {
  mapStateToProps,
  CaseStudyForm as Form
}

export default connect(mapStateToProps)(CaseStudyForm);
