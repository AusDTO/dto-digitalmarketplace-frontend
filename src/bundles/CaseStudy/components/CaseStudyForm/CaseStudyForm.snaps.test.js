// FIXME This is a workaround.
// @see https://github.com/facebook/react/issues/7386
jest.mock('react-dom');
//jest.mock('react-router');

import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router';
import renderer from 'react-test-renderer';

import CaseStudyForm, { Textfield } from './CaseStudyForm';
import StepOne from './StepOne';
import StepTwo from './StepTwo';
import sampleState from './CaseStudyForm.json';
import createStore from '../../redux/create';

test('CaseStudyForm renders', () => {
  let store = createStore(Object.assign({}, { _serverContext: {} }))
  const component = renderer.create(
    <BrowserRouter>
      <Provider store={store}>
        <CaseStudyForm />
      </Provider>
    </BrowserRouter>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('CaseStudyForm renders with form_options', () => {
  const form_options = {
    csrf_token: 'sometoken',
    action: '/foo/bar'
  }
  let store = createStore(Object.assign({}, { _serverContext: {}, form_options }))
  const component = renderer.create(
    <BrowserRouter>
      <Provider store={store}>
        <CaseStudyForm />
      </Provider>
    </BrowserRouter>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('CaseStudyForm renders with populated fields', () => {
  let store = createStore(Object.assign({}, { _serverContext: {} }, sampleState))
  const component = renderer.create(
    <BrowserRouter>
      <Provider store={store}>
        <CaseStudyForm />
      </Provider>
    </BrowserRouter>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

