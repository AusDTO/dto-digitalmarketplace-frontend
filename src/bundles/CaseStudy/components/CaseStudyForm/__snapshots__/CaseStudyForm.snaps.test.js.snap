exports[`test CaseStudyForm renders 1`] = `
<div>
  <form
    style={
      Object {
        "display": "none",
      }
    }
    tabIndex="-1" />
  <div
    className="layout">
    <header>
      <h1>
        Add
         case study
      </h1>
      <p>
        Show the range of skills and experience you can provide by completing the form below.
      </p>
    </header>
    <aside
      className="sidebar">
      <nav
        className="local-nav nav-collapse">
        <ul>
          <li>
            <a
              className={undefined}
              href="/"
              onClick={[Function]}
              role={undefined}>
              Add case study
            </a>
          </li>
          <li>
            <a
              className={undefined}
              href="/reference"
              onClick={[Function]}
              role={undefined}>
              Add reference
            </a>
          </li>
        </ul>
      </nav>
    </aside>
    <article
      className="content-main"
      role="main">
      <span />
      <form
        action={undefined}
        id="casestudy__create"
        method="post"
        onReset={[Function]}
        onSubmit={[Function]}>
        <div
          className="field">
          <label
            className="question-heading"
            htmlFor="title">
            Give your case study a title
          </label>
          <input
            aria-describedby={undefined}
            className=""
            id="title"
            name="title"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            type="text"
            value="" />
        </div>
        <div
          className="field">
          <label
            className="question-heading"
            htmlFor="client">
            Who was the client?
          </label>
          <input
            aria-describedby={undefined}
            className=""
            id="client"
            name="client"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            type="text"
            value="" />
        </div>
        <div
          className="field">
          <label
            className="question-heading"
            htmlFor="timeframe">
            What was the time frame?
          </label>
          <p
            className="hint"
            id="timeframe-hint">
            For example,  January 2016 â€” June 2016
          </p>
          <input
            aria-describedby="timeframe-hint"
            className=""
            id="timeframe"
            name="timeframe"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            type="text"
            value="" />
        </div>
        <div
          className="field">
          <label
            htmlFor="opportunity">
            Outline the problem or opportunity
          </label>
          <p
            className="hint"
            id="opportunity-hint">
            Describe the project goal and any relevant background information.
          </p>
          <div>
            <textarea
              aria-describedby="opportunity-hint"
              className=""
              defaultValue=""
              id="opportunity"
              name="opportunity"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]} />
            <span
              className="word-count-counter">
              200 words remaining
            </span>
          </div>
        </div>
        <div
          className="field">
          <label
            htmlFor="approach">
            Describe your approach
          </label>
          <p
            className="hint"
            id="approach-hint">
            How did your capabilities and methods contribute to achieving the project goals?
          </p>
          <div>
            <textarea
              aria-describedby="approach-hint"
              className=""
              defaultValue=""
              id="approach"
              name="approach"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]} />
            <span
              className="word-count-counter">
              200 words remaining
            </span>
          </div>
        </div>
        <div
          className="field">
          <label
            htmlFor="outcome">
            What was the outcome?
          </label>
          <p
            className="hint"
            id="outcome-hint">
            List the key achievements of this project.
          </p>
          <div>
            <div
              className="list-entry row">
              <div
                className="col-xs-12 col-sm-9">
                <label
                  className="text-box-number-label"
                  htmlFor="outcome">
                  1
                  .
                </label>
                <input
                  aria-describedby="outcome-hint"
                  className="text-box "
                  defaultValue=""
                  id="outcome"
                  name="outcome[]"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  type="text" />
              </div>
            </div>
            <div
              className="list-entry row">
              <div
                className="col-xs-12 col-sm-9">
                <label
                  className="text-box-number-label"
                  htmlFor="outcome-1">
                  2
                  .
                </label>
                <input
                  aria-describedby="outcome-hint"
                  className="text-box "
                  defaultValue=""
                  id="outcome-1"
                  name="outcome[]"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  type="text" />
              </div>
              <button
                className="button-secondary col-xs-12 col-sm-3"
                onClick={[Function]}
                type="button">
                remove 
                <span
                  className="visuallyhidden">
                  number 
                  2
                </span>
              </button>
            </div>
            <button
              className="button-secondary"
              onClick={[Function]}
              type="button">
              Add another row
            </button>
          </div>
        </div>
        <div
          className="field">
          <label
            htmlFor="projectLinks">
            Project links
          </label>
          <p
            className="hint"
            id="projectLinks-hint">
            Link to any supporting material for your case study. This can include a case study on your  website, case study video or the live project.
          </p>
          <div>
            <div
              className="list-entry row">
              <div
                className="col-xs-12 col-sm-9">
                <label
                  className="text-box-number-label"
                  htmlFor="projectLinks">
                  1
                  .
                </label>
                <input
                  aria-describedby="projectLinks-hint"
                  className="text-box "
                  defaultValue=""
                  id="projectLinks"
                  name="projectLinks[]"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  type="text" />
              </div>
            </div>
            <div
              className="list-entry row">
              <div
                className="col-xs-12 col-sm-9">
                <label
                  className="text-box-number-label"
                  htmlFor="projectLinks-1">
                  2
                  .
                </label>
                <input
                  aria-describedby="projectLinks-hint"
                  className="text-box "
                  defaultValue=""
                  id="projectLinks-1"
                  name="projectLinks[]"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  type="text" />
              </div>
              <button
                className="button-secondary col-xs-12 col-sm-3"
                onClick={[Function]}
                type="button">
                remove 
                <span
                  className="visuallyhidden">
                  number 
                  2
                </span>
              </button>
            </div>
            <button
              className="button-secondary"
              onClick={[Function]}
              type="button">
              Add another row
            </button>
          </div>
        </div>
        <input
          onClick={[Function]}
          role="button"
          type="submit"
          value="Save & Continue" />
      </form>
    </article>
  </div>
</div>
`;

exports[`test CaseStudyForm renders with form_options 1`] = `
<div>
  <form
    style={
      Object {
        "display": "none",
      }
    }
    tabIndex="-1" />
  <div
    className="layout">
    <header>
      <h1>
        Add
         case study
      </h1>
      <p>
        Show the range of skills and experience you can provide by completing the form below.
      </p>
    </header>
    <aside
      className="sidebar">
      <nav
        className="local-nav nav-collapse">
        <ul>
          <li>
            <a
              className={undefined}
              href="/"
              onClick={[Function]}
              role={undefined}>
              Add case study
            </a>
          </li>
          <li>
            <a
              className={undefined}
              href="/reference"
              onClick={[Function]}
              role={undefined}>
              Add reference
            </a>
          </li>
        </ul>
      </nav>
    </aside>
    <article
      className="content-main"
      role="main">
      <span />
      <form
        action="/foo/bar"
        id="casestudy__create"
        method="post"
        onReset={[Function]}
        onSubmit={[Function]}>
        <input
          id="csrf_token"
          name="csrf_token"
          type="hidden"
          value="sometoken" />
        <div
          className="field">
          <label
            className="question-heading"
            htmlFor="title">
            Give your case study a title
          </label>
          <input
            aria-describedby={undefined}
            className=""
            id="title"
            name="title"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            type="text"
            value="" />
        </div>
        <div
          className="field">
          <label
            className="question-heading"
            htmlFor="client">
            Who was the client?
          </label>
          <input
            aria-describedby={undefined}
            className=""
            id="client"
            name="client"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            type="text"
            value="" />
        </div>
        <div
          className="field">
          <label
            className="question-heading"
            htmlFor="timeframe">
            What was the time frame?
          </label>
          <p
            className="hint"
            id="timeframe-hint">
            For example,  January 2016 â€” June 2016
          </p>
          <input
            aria-describedby="timeframe-hint"
            className=""
            id="timeframe"
            name="timeframe"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            type="text"
            value="" />
        </div>
        <div
          className="field">
          <label
            htmlFor="opportunity">
            Outline the problem or opportunity
          </label>
          <p
            className="hint"
            id="opportunity-hint">
            Describe the project goal and any relevant background information.
          </p>
          <div>
            <textarea
              aria-describedby="opportunity-hint"
              className=""
              defaultValue=""
              id="opportunity"
              name="opportunity"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]} />
            <span
              className="word-count-counter">
              200 words remaining
            </span>
          </div>
        </div>
        <div
          className="field">
          <label
            htmlFor="approach">
            Describe your approach
          </label>
          <p
            className="hint"
            id="approach-hint">
            How did your capabilities and methods contribute to achieving the project goals?
          </p>
          <div>
            <textarea
              aria-describedby="approach-hint"
              className=""
              defaultValue=""
              id="approach"
              name="approach"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]} />
            <span
              className="word-count-counter">
              200 words remaining
            </span>
          </div>
        </div>
        <div
          className="field">
          <label
            htmlFor="outcome">
            What was the outcome?
          </label>
          <p
            className="hint"
            id="outcome-hint">
            List the key achievements of this project.
          </p>
          <div>
            <div
              className="list-entry row">
              <div
                className="col-xs-12 col-sm-9">
                <label
                  className="text-box-number-label"
                  htmlFor="outcome">
                  1
                  .
                </label>
                <input
                  aria-describedby="outcome-hint"
                  className="text-box "
                  defaultValue=""
                  id="outcome"
                  name="outcome[]"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  type="text" />
              </div>
            </div>
            <div
              className="list-entry row">
              <div
                className="col-xs-12 col-sm-9">
                <label
                  className="text-box-number-label"
                  htmlFor="outcome-1">
                  2
                  .
                </label>
                <input
                  aria-describedby="outcome-hint"
                  className="text-box "
                  defaultValue=""
                  id="outcome-1"
                  name="outcome[]"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  type="text" />
              </div>
              <button
                className="button-secondary col-xs-12 col-sm-3"
                onClick={[Function]}
                type="button">
                remove 
                <span
                  className="visuallyhidden">
                  number 
                  2
                </span>
              </button>
            </div>
            <button
              className="button-secondary"
              onClick={[Function]}
              type="button">
              Add another row
            </button>
          </div>
        </div>
        <div
          className="field">
          <label
            htmlFor="projectLinks">
            Project links
          </label>
          <p
            className="hint"
            id="projectLinks-hint">
            Link to any supporting material for your case study. This can include a case study on your  website, case study video or the live project.
          </p>
          <div>
            <div
              className="list-entry row">
              <div
                className="col-xs-12 col-sm-9">
                <label
                  className="text-box-number-label"
                  htmlFor="projectLinks">
                  1
                  .
                </label>
                <input
                  aria-describedby="projectLinks-hint"
                  className="text-box "
                  defaultValue=""
                  id="projectLinks"
                  name="projectLinks[]"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  type="text" />
              </div>
            </div>
            <div
              className="list-entry row">
              <div
                className="col-xs-12 col-sm-9">
                <label
                  className="text-box-number-label"
                  htmlFor="projectLinks-1">
                  2
                  .
                </label>
                <input
                  aria-describedby="projectLinks-hint"
                  className="text-box "
                  defaultValue=""
                  id="projectLinks-1"
                  name="projectLinks[]"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  type="text" />
              </div>
              <button
                className="button-secondary col-xs-12 col-sm-3"
                onClick={[Function]}
                type="button">
                remove 
                <span
                  className="visuallyhidden">
                  number 
                  2
                </span>
              </button>
            </div>
            <button
              className="button-secondary"
              onClick={[Function]}
              type="button">
              Add another row
            </button>
          </div>
        </div>
        <input
          onClick={[Function]}
          role="button"
          type="submit"
          value="Save & Continue" />
      </form>
    </article>
  </div>
</div>
`;

exports[`test CaseStudyForm renders with populated fields 1`] = `
<div>
  <form
    style={
      Object {
        "display": "none",
      }
    }
    tabIndex="-1" />
  <div
    className="layout">
    <header>
      <h1>
        Add
         case study
      </h1>
      <p>
        Show the range of skills and experience you can provide by completing the form below.
      </p>
    </header>
    <aside
      className="sidebar">
      <nav
        className="local-nav nav-collapse">
        <ul>
          <li>
            <a
              className={undefined}
              href="/"
              onClick={[Function]}
              role={undefined}>
              Add case study
            </a>
          </li>
          <li>
            <a
              className={undefined}
              href="/reference"
              onClick={[Function]}
              role={undefined}>
              Add reference
            </a>
          </li>
        </ul>
      </nav>
    </aside>
    <article
      className="content-main"
      role="main">
      <span />
      <form
        action="https://httpbin.org/post"
        id="casestudy__create"
        method="post"
        onReset={[Function]}
        onSubmit={[Function]}>
        <div
          className="field">
          <label
            className="question-heading"
            htmlFor="title">
            Give your case study a title
          </label>
          <input
            aria-describedby={undefined}
            className=""
            id="title"
            name="title"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            type="text"
            value="" />
        </div>
        <div
          className="field">
          <label
            className="question-heading"
            htmlFor="client">
            Who was the client?
          </label>
          <input
            aria-describedby={undefined}
            className=""
            id="client"
            name="client"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            type="text"
            value="The Client Name" />
        </div>
        <div
          className="field">
          <label
            className="question-heading"
            htmlFor="timeframe">
            What was the time frame?
          </label>
          <p
            className="hint"
            id="timeframe-hint">
            For example,  January 2016 â€” June 2016
          </p>
          <input
            aria-describedby="timeframe-hint"
            className=""
            id="timeframe"
            name="timeframe"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            type="text"
            value="January 2016 â€” June 2016" />
        </div>
        <div
          className="field">
          <label
            htmlFor="opportunity">
            Outline the problem or opportunity
          </label>
          <p
            className="hint"
            id="opportunity-hint">
            Describe the project goal and any relevant background information.
          </p>
          <div>
            <textarea
              aria-describedby="opportunity-hint"
              className=""
              defaultValue="The opportunity"
              id="opportunity"
              name="opportunity"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]} />
            <span
              className="word-count-counter">
              198 words remaining
            </span>
          </div>
        </div>
        <div
          className="field">
          <label
            htmlFor="approach">
            Describe your approach
          </label>
          <p
            className="hint"
            id="approach-hint">
            How did your capabilities and methods contribute to achieving the project goals?
          </p>
          <div>
            <textarea
              aria-describedby="approach-hint"
              className=""
              defaultValue="The approach"
              id="approach"
              name="approach"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]} />
            <span
              className="word-count-counter">
              198 words remaining
            </span>
          </div>
        </div>
        <div
          className="field">
          <label
            htmlFor="outcome">
            What was the outcome?
          </label>
          <p
            className="hint"
            id="outcome-hint">
            List the key achievements of this project.
          </p>
          <div>
            <div
              className="list-entry row">
              <div
                className="col-xs-12 col-sm-9">
                <label
                  className="text-box-number-label"
                  htmlFor="outcome">
                  1
                  .
                </label>
                <input
                  aria-describedby="outcome-hint"
                  className="text-box "
                  defaultValue="Outcome 1"
                  id="outcome"
                  name="outcome[]"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  type="text" />
              </div>
            </div>
            <div
              className="list-entry row">
              <div
                className="col-xs-12 col-sm-9">
                <label
                  className="text-box-number-label"
                  htmlFor="outcome-1">
                  2
                  .
                </label>
                <input
                  aria-describedby="outcome-hint"
                  className="text-box "
                  defaultValue="Outcome 2"
                  id="outcome-1"
                  name="outcome[]"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  type="text" />
              </div>
              <button
                className="button-secondary col-xs-12 col-sm-3"
                onClick={[Function]}
                type="button">
                remove 
                <span
                  className="visuallyhidden">
                  number 
                  2
                </span>
              </button>
            </div>
            <button
              className="button-secondary"
              onClick={[Function]}
              type="button">
              Add another row
            </button>
          </div>
        </div>
        <div
          className="field">
          <label
            htmlFor="projectLinks">
            Project links
          </label>
          <p
            className="hint"
            id="projectLinks-hint">
            Link to any supporting material for your case study. This can include a case study on your  website, case study video or the live project.
          </p>
          <div>
            <div
              className="list-entry row">
              <div
                className="col-xs-12 col-sm-9">
                <label
                  className="text-box-number-label"
                  htmlFor="projectLinks">
                  1
                  .
                </label>
                <input
                  aria-describedby="projectLinks-hint"
                  className="text-box "
                  defaultValue=""
                  id="projectLinks"
                  name="projectLinks[]"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  type="text" />
              </div>
            </div>
            <div
              className="list-entry row">
              <div
                className="col-xs-12 col-sm-9">
                <label
                  className="text-box-number-label"
                  htmlFor="projectLinks-1">
                  2
                  .
                </label>
                <input
                  aria-describedby="projectLinks-hint"
                  className="text-box "
                  defaultValue=""
                  id="projectLinks-1"
                  name="projectLinks[]"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  type="text" />
              </div>
              <button
                className="button-secondary col-xs-12 col-sm-3"
                onClick={[Function]}
                type="button">
                remove 
                <span
                  className="visuallyhidden">
                  number 
                  2
                </span>
              </button>
            </div>
            <button
              className="button-secondary"
              onClick={[Function]}
              type="button">
              Add another row
            </button>
          </div>
        </div>
        <input
          onClick={[Function]}
          role="button"
          type="submit"
          value="Save & Continue" />
      </form>
    </article>
  </div>
</div>
`;
